code/game/objects/items/toys.dm

```diff
@@ -691,19 +689,49 @@
 	if(cards.len == 0)
 		to_chat(user, "<span class='warning'>There are no more cards to draw!</span>")
 		return
-	var/obj/item/toy/cards/singlecard/H = new/obj/item/toy/cards/singlecard(user.loc)
-	if(holo)
-		holo.spawned += H // track them leaving the holodeck
-	choice = cards[1]
-	H.cardname = choice
-	H.parentdeck = src
-	var/O = src
-	H.apply_card_vars(H,O)
-	src.cards.Cut(1,2)
-	H.pickup(user)
-	user.put_in_hands(H)
-	user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
-	update_icon()
+	if (drawnumber==1)
+		var/obj/item/toy/cards/singlecard/C = new/obj/item/toy/cards/singlecard(user.loc)
+		choice = cards[1]
+		user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
+		C.cardname = choice
+		if(holo)
+			holo.spawned += C // track them leaving the holodeck
+		C.parentdeck = src
+		var/O = src
+		C.apply_card_vars(C,O)
+		C.deckstyle=deckstyle
+		src.cards.Cut(1,2)
+		C.pickup(user)
+		user.put_in_hands(C)
```

Comment:
Is there a reason to call pickup manually here? put_in_hands should call it automatically AFAIK

Reply:
pickup adds messages "you pick up xyz" /shrug I can test it without

Reply:
works fine without